internal/utils/akka/src/test/java/org/eclipse/ditto/internal/utils/akka/controlflow/TimeMeasuringFlowTest.java

```diff
@@ -74,53 +77,55 @@ public void tearDown() {
-            return x;
-        });
+        final Duration sleepDuration = Duration.ofMillis(1000);
+        final Flow<String, String, NotUsed> flowThatNeedsSomeTime = Flow.<String>create()
+                .delay(sleepDuration, DelayOverflowStrategy.backpressure())
+                .map(x -> {
+                    // Flow.delay is not accurate
+                    flowCounter.incrementAndGet();
+                    return x;
+                });
 
         final PreparedTimer timer = DittoMetrics.timer("test-time-measuring-flow");
         final PreparedTimer timerMock = mock(PreparedTimer.class);
         when(timerMock.start()).thenAnswer(AdditionalAnswers.delegatesTo(timer));
-        final Sink<Duration, TestSubscriber.Probe<Duration>> sink = TestSink.<Duration>probe(system);
-        final Pair<TestSubscriber.Probe<Duration>, Sink<Duration, NotUsed>> sinkPair = sink.preMaterialize(system);
-        final TestSubscriber.Probe<Duration> durationProbe = sinkPair.first();
+
+        final Queue<Duration> durations = new ConcurrentLinkedQueue<>();
         new TestKit(system) {{
+            final int numberOfRepetitions = 10;
             Source.repeat("Test")
-                    .via(TimeMeasuringFlow.measureTimeOf(flowThatNeedsSomeTime, timerMock, sinkPair.second()))
+                    .take(10)
+                    .via(TimeMeasuringFlow.measureTimeOf(flowThatNeedsSomeTime, timerMock, 99, i -> {}, durations::add))
                     .via(flowThatNeedsSomeTime) // This should not influence the time measuring above
                     .to(testSink)
                     .run(system);
 
-            final int numberOfRepetitions = 10;
             final ArrayList<String> expectedResults = new ArrayList<>(numberOfRepetitions);
             for (int i = 0; i < numberOfRepetitions; i++) {
                 expectedResults.add("Test");
             }
-            durationProbe.request(numberOfRepetitions);
             sinkProbe.request(numberOfRepetitions);
             sinkProbe.expectNextN(CollectionConverters.asScala(expectedResults).toSeq());
-            final List<Duration> durations =
-                    CollectionConverters.asJava(durationProbe.expectNextN(numberOfRepetitions));
 
+            if (durations.size() < numberOfRepetitions) {
+                Thread.sleep(sleepDuration.toMillis());
+            }
             assertThat(durations).hasSize(numberOfRepetitions);
+
             final double averageDurationInNanos = durations.stream()
                     .mapToLong(Duration::toNanos)
                     .average()
                     .orElseThrow();
             /*
-             * Needs to be lesser than sleep duration to verify that potential inaccuracies are not mixed up with
-             * what this test you ensure.
+             * Needs to be lesser than sleep duration to verify that potential inaccuracies are not mixed up.
+             * Allow 25% inaccuracy.
```

Comment:
Why do we allow such a high inaccuracy? This means that this test allows 250ms of inaccuracy which seems to make this time measuring pointless.

Reply:
This test applies delay to all elements, which means time taken to process previous elements are added to subsequent timers. I will make the delay per element. It should make the measurement more accurate up to 1s. Sub-1s measurements are inaccurate because scheduling delay is still measured.